src/Blazor.Gitter.Core/Components/Shared/ChatRoomList.razor.cs

```diff
@@ -1,29 +1,55 @@
-﻿using Microsoft.AspNetCore.Components;
+﻿using Blazor.Gitter.Library;
+using Microsoft.AspNetCore.Components;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Timers;
 
 namespace Blazor.Gitter.Core.Components.Shared
 {
-    public class ChatRoomListBase : ComponentBase
+    public class ChatRoomListBase : ComponentBase, IDisposable
     {
         [Inject] internal IAppState State { get; set; }
+        [Inject] IChatApi GitterApi { get; set; }
+        public Action GetRooms { get; private set; }
 
-        protected bool HasRooms;
-
+        private Timer ChatRoomTimer;
+        private const int CHATROOMUPDATETIME = 10000;

```

Comment:
Timer to update the chat room list - if we could find a way to get notifications from Gitter....that would help.